tsfc/driver.py

```diff
@@ -137,7 +138,8 @@ def compile_integral(integral_data, form_data, prefix, parameters,
         mode = pick_mode(params["mode"])
         mode_irs.setdefault(mode, collections.OrderedDict())
 
-        integrand = ufl.replace(integral.integrand(), form_data.function_replace_map)
+        # integrand = ufl.replace(integral.integrand(), form_data.function_replace_map)
+        integrand = integral.integrand()
```

Comment:
This was originally added because of some DOLFIN `Expression` issue, and this draft change undoes that. So it likely breaks some cases of using DOLFIN with TSFC, but I did not have time to debug how to make every case work.

Reply:
I think I fixed the failing that this caused for some other context via UFL changes a while ago.  So I reverted these bits and am running the Firedrake test suite to see if everything still works.

Reply:
Yeah, that seemed to work.

Reply:
That's good!